firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestoreSettings.java

```diff
@@ -105,14 +119,55 @@ public Builder setPersistenceEnabled(boolean value) {
      *
      * @return A settings object on which the cache size is configured as specified by the given
      *     {@code value}.
+     * @deprecated Use {@link
+     *     FirebaseFirestoreSettings.Builder#setLocalCacheSettings(LocalCacheSettings)} to configure
+     *     SDK cache instead.
      */
     @NonNull
+    @Deprecated
     public Builder setCacheSizeBytes(long value) {
+      if (cacheSettings != null) {
+        throw new IllegalStateException(
+            "New cache config API setLocalCacheSettings() is already used.");
+      }
       if (value != CACHE_SIZE_UNLIMITED && value < MINIMUM_CACHE_BYTES) {
         throw new IllegalArgumentException(
             "Cache size must be set to at least " + MINIMUM_CACHE_BYTES + " bytes");
       }
       this.cacheSizeBytes = value;
+      this.usedLegacyCacheSettings = true;
+      return this;
+    }
+
+    /**
+     * Specifies the cache used by the SDK. Available options are {@link PersistentCacheSettings}
+     * and {@link MemoryCacheSettings}, each with different configuration options.
+     *
+     * <p>When unspecified, {@link PersistentCacheSettings} will be used by default.
+     *
+     * <p>NOTE: Calling this setter and {@code setPersistenceEnabled()} or @{code
+     * setCacheSizeBytes()} at the same time will throw an exception during SDK initialization.
+     * Instead, use the configuration in the {@link PersistentCacheSettings} object to specify the
+     * cache size.
+     *
+     * @return A settings object on which the cache settings is configured as specified by the given
+     *     {@code settings}.
+     */
+    @NonNull
+    public Builder setLocalCacheSettings(@NonNull LocalCacheSettings settings) {
```

Comment:
All other properties have corresponding get method. I would expect this to have a default value of `PersistentCacheSettings`, just like other properties have default values.

Reply:
This does avoid problems and can easily be added later if we change our minds.